librad/src/net/peer.rs

```diff
@@ -467,22 +467,18 @@ where
 /// `refs/remotes/<origin>/<path>`
 fn urn_context(local_peer_id: &PeerId, urn: Either<RadUrn, Originates<RadUrn>>) -> RadUrn {
     match urn {
-        Either::Left(urn) => urn,
+        Either::Left(urn) => RadUrn {
+            path: qualify_path(&urn.path),
```

Comment:
I retract my approval here: if we have `Left("rad:git:XXX/refs/rad/id")`, the result of calling `qualify_path` will be "rad:git:XXX/rad/id". I think we rather want the qualified form.

Reply:
:facepalm: right, so it should leave `refs/rad` paths alone?

Reply:
Ya it's because we have `refs/remotes/XXX/{heads,rad}`. So:

```
rel/ative -> refs/remotes/XXX/heads/rel/ative
refs/abs/olute -> refs/remotes/XXX/refs/abs/olute # even though nonsensical
rel/alive -> refs/heads/rel/ative
refs/abs/olute -> refs/abs/olute # even though nonsensical
```

Reply:
https://github.com/radicle-dev/radicle-link/pull/401/commits/183f3176dac3415072332287024e5df47fddd1bd

Reply:
I do not, btw, recall what the purpose of `local_peer_id` is -- if it originates from ourselves, we should just return `Stale`

Reply:
Well `urn_context` is used in two places:
* [`git_fetch`](https://github.com/radicle-dev/radicle-link/blob/fintan/urn-context-patch/librad/src/net/peer.rs#L421)
* [`git_has`](https://github.com/radicle-dev/radicle-link/blob/fintan/urn-context-patch/librad/src/net/peer.rs#L448)

`git_has` is used in `ask` which is called in two places: when we `put` and when gossip receives a `Want`. So I think if the `Want` ends up having the same origin as the current peer id, then the `RadUrn` shouldn't become `remotes` path.

I'll see if I can make this clearer :thinking:

Reply:
How's this look https://github.com/radicle-dev/radicle-link/pull/401/commits/16d1829ac7c4376249eec8f9a2a0fd4e5aec380f?

Reply:
:+1: